### PR TITLE
chore: migrate npm package logic

### LINKED ISSUE #1: Create mento-core-ts repo with deployment code added
**Description**
We need a repo to contain the logic needed to generate the typescript types for the mento-core contracts and to allow us to push these types as an NPM package to the mento-protocol registry.

**Acceptance Criteria**

- [x] A new repo `mento-core-ts` is created with the same policies as mento-core
- [x] The code to generate TS types is moved from mento-core and into this repo
- [x] Typescript type generation works as before
- [x] The workflow to publish the NPM package is moved from mento-core and into this repo
- [x] NPM package publishing can be triggered manually, by members of the team, and works as expected
- [x] A doc is added to the Eng wiki detailing the steps someone should take publish a new version of mento-core-ts
- [x] A readme is added at the root of the repo that describes how a client application can make use of the package

### PR DESCRIPTION
**Pull request by ninabarbakadze**
### Description
Moved mento core contracts type generation and NPM deployment logic from mento-core repo.

### Other changes

n/a

### Tested

Successfully published the [package](https://www.npmjs.com/package/@mento-protocol/mento-core-ts?activeTab=readme) on push 

In order to publish manually and test, we'll have to merge the PR

### Related issues

- Fixes #1 

### Backwards compatibility

n/a

### Documentation

[NPM docs](https://www.notion.so/Mento-Core-NPM-Package-393ea86a8a824a3183bdec34b9abb250)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,34 @@
...
+cd mento-tooling
+
+# Install dev dependencies with yarn
+yarn
+
+# Install mento-core submodule dependency with forge
+forge install
+
+# Compile the mento-core dependency contracts with forge
+forge forge build -c ./lib/mento-core/contracts
```
Comment (ninabarbakadze) [inline README.md]: Not sure if we need to build contracts in the dependency, npm yml workflow repeats the same step

File: README.md [hunk 2]
```diff
@@ -1 +1,34 @@
-# mento-tooling
\ No newline at end of file
+# Mento Core TS
+
+This repo contains the source code of the Mento core contracts type generation and NPM package deployment.
```
Comment (bowd) [inline README.md]: This should be written as if the audience is somebody looking to interact with the mento protocol via the generated types and factories. I understand why you phrased it like this because that's how we've been talking about it internally but from the perspective of somebody external that's weird because all repos with js/ts libraries do NPM package deployment.

Let's phrase more around the lines of "this package provides auto-generated typescript types and ethers.js factories for interacting with the Mento Protocol"

File: README.md (hunk 1 above)
Comment (bowd) [reply README.md]: Not sure what you mean by this, can you rephrase it?
Side note there's `forge` written two times here.

**Review by bowd — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  ci:
+    name: Lint & Test
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: "Check out the repo"
+        uses: "actions/checkout@v3"
+        with:
+          submodules: "recursive"
+
+      - name: "Install Foundry"
```
Comment (philbow61) [inline .github/workflows/ci.yml]: wondering why we are installing foundry in the ci if its not used currently

File: .github/workflows/publish_npm.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          node-version: "16"
+          registry-url: "https://registry.npmjs.org"
+
+      - name: "Install the Node.js dependencies"
+        run: "yarn install --immutable"
+
+      - name: "Build the contracts"
+        run: |
+          forge --version
+          forge build -c ./lib/mento-core/contracts --sizes
```
Comment (philbow61) [inline .github/workflows/publish_npm.yml]: maybe we could remove the path here and instead change the foundry.toml

File: foundry.toml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+[profile.default]
+src = "contracts"
```
Comment (philbow61) [inline foundry.toml]:
```suggestion
src = "lib/mento-core/contracts"
```

**Review by philbow61 — CHANGES_REQUESTED** [review]
LGTM 👍 but left some minor questions

File: README.md (hunk 2 above)
Comment (ninabarbakadze) [reply README.md]: yeah, I made the mistake of writing the readme for the repo.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ninabarbakadze) [reply .github/workflows/ci.yml]: yeah gonna remove it from ci 👍🏼

File: .github/workflows/publish_npm.yml (hunk 4 above)
Comment (ninabarbakadze) [reply .github/workflows/publish_npm.yml]: that'd be cleaner, let's do that

**Review by philbow61 — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    "img"
+  ],
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "author": {
+    "name": "mento-protocol"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/mento-protocol/mento-core"
```
Comment (bowd) [inline package.json:18]: Should be `mento-core-ts`

Comment (bowd) [inline package.json]: Should be `mento-core-ts`

**Review by bowd — APPROVED** [review]
Just one small URL change needed, but approving this already :shipit:
